Author: aszlig <aszlig@nix.build>
Date:   Sat Jun 23 14:25:29 2018 +0200

    Allow to purchase gems using gold from menu
    
    The icon was already there so far but instead of getting to the
    buyModal, it instead showed a modal to purchase gems for real money.
    
    All of our subscriptions are free so we instead reuse the same icon to
    purchase gems using gold.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: buy-gems-via-menu.patch

diff --git a/website/client/components/header/menu.vue b/website/client/components/header/menu.vue
index 3eb2ed0c9..514788bde 100644
--- a/website/client/components/header/menu.vue
+++ b/website/client/components/header/menu.vue
@@ -59,7 +59,7 @@ div
           .top-menu-icon.svg-icon(v-html="icons.hourglasses", v-b-tooltip.hover.bottom="$t('mysticHourglassesTooltip')")
           span {{ userHourglasses }}
         .item-with-icon
-          .top-menu-icon.svg-icon.gem(v-html="icons.gem", @click='showBuyGemsModal("gems")', v-b-tooltip.hover.bottom="$t('gems')")
+          .top-menu-icon.svg-icon.gem(v-html="icons.gem", @click='showBuyGemsModal', v-b-tooltip.hover.bottom="$t('gems')")
           span {{userGems}}
         .item-with-icon.gold
           .top-menu-icon.svg-icon(v-html="icons.gold", v-b-tooltip.hover.bottom="$t('gold')")
@@ -345,6 +345,7 @@ import profileModal from '../userMenu/profileModal';
 import userDropdown from './userDropdown';
 
 import reportFlagModal from '../chat/reportFlagModal';
+import getItemInfo from 'common/script/libs/getItemInfo';
 
 export default {
   components: {
@@ -405,11 +406,10 @@ export default {
     openPartyModal () {
       this.$root.$emit('bv::show::modal', 'create-party-modal');
     },
-    showBuyGemsModal (startingPage) {
-      this.$store.state.gemModalOptions.startingPage = startingPage;
-      this.$root.$emit('bv::show::modal', 'buy-gems', {alreadyTracked: true});
+    showBuyGemsModal () {
+      let gemItem = getItemInfo(this.user, 'gem');
+      this.$root.$emit('buyModal::showItem', gemItem);
     },
-
   },
 };
 </script>
